init.d/seedrng.in

```diff
@@ -0,0 +1,46 @@
+#!@SBINDIR@/openrc-run
+# Copyright (c) 2007-2022 The OpenRC Authors.
+# See the Authors file at the top-level directory of this distribution and
+# https://github.com/OpenRC/openrc/blob/HEAD/AUTHORS
+#
+# This file is part of OpenRC. It is subject to the license terms in
+# the LICENSE file found in the top-level directory of this
+# distribution and at https://github.com/OpenRC/openrc/blob/HEAD/LICENSE
+# This file may not be copied, modified, propagated, or distributed
+# except according to the terms contained in the LICENSE file.
+
+description="Initializes the random number generator."
+
+depend()
+{
+	after clock
+	need localmount
+	keyword -docker -jail -lxc -openvz -prefix -systemd-nspawn
+	provide urandom
+}
+
+seedrng_options()
+{
+	[ -n "${lock_file}" ] &&
+		echo "--lock-file \"${lock_file}\""
+	[ -n "${seed_dir}" ] &&
+		echo "--seed-dir \"${seed_dir}\""
+	rc_yesno "${skip_credit}" &&
+		echo "--skip-credit"
+}
+
+start()
+{
+	ebegin "Seeding random number generator"
+	seedrng $(seedrng_options)
+	eend $? "Error seeding random number generator"
+	return 0
+}
+
+stop()
+{
+	ebegin "Seeding random number generator"
```

Comment:
Ah, one more nit: maybe change this text to "Saving random number generator seed". Yes, it's the same operation as start(), but it serves a slightly different purpose here.